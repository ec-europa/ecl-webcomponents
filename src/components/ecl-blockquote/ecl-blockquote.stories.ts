const getArgs = () => {
  return {
    author: 'President Juncker',
    quote: `An interconnected grid will help deliver the ultimate goal of the Energy Union, 
      to make sure affordable, secure and sustainable energy, and also growth across the EU.`,
    language: 'en',
    image: false,
    imageAlt: 'Image alternative text',
  };
};

const getArgTypes = () => {
  return {
    author: {
      type: { name: 'string' },
      description: "Author of the quote"
    },
    quote: {
      type: { name: 'string' },
      description: "Text of the quote"
    },
    language: {
      type: { name: 'string' },
      description: "Localization"
    },
    image: {
      name: ' with image',
      type: { name: 'boolean'},
      description: 'Add an image',
    },
    imageAlt: {
      name: 'alternative text for the image',
      type: { name: 'string' },
      description: "To increase accessibility",
    },
  };
};

export default {
  title: 'Components/blockquote',
};

const Template = (args) => 
`<ecl-blockquote
  author="${args.author}"
  color-mode="${args.color_mode}"
  theme="${args.theme}"
  image="${args.image ? 'https://inno-ecl.s3.amazonaws.com/media/examples/example-image-square.jpg' : ''}"
  image-alt="${args.imageAlt}"
  language="${args.language}"
>
  ${args.quote}
</ecl-blockquote>`;

export const Blockquote = Template.bind({});
Blockquote.storyName = 'default';
Blockquote.args = getArgs();
Blockquote.argTypes = getArgTypes();