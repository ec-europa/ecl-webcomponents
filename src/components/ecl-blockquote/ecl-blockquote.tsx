import { Component, Prop, h } from '@stencil/core';

@Component({
  tag: 'ecl-blockquote',
  styleUrls: {
    ec: './build/styles/ecl-blockquote-ec.css',
    eu: './build/styles/ecl-blockquote-eu.css',
  },
  shadow: false,
  scoped: true,
})
export class EclBlockquote {
  @Prop() styleClass: string = '';
  @Prop({ mutable: true }) theme: string;
  @Prop() image: string;
  @Prop() author: string;
  @Prop() imageAlt: string;
  @Prop() colorMode: string;
  @Prop() language: string = 'en';

  getClass(): string {
    const styleClasses = [
      `ecl-blockquote`,
      this.styleClass
    ];

    if (this.colorMode) {
      styleClasses.push(`ecl-color-mode--${this.colorMode}`)
    }

    return styleClasses.join(' ');
  }

  componentWillLoad() {
    this.theme = document.documentElement.getAttribute('data-ecl-theme') ?? (this.theme || 'ec');
  }

  render() {
    return (
      <figure class={this.getClass()}>
      { this.image ?
        <ecl-picture
          styleClass={`ecl-blockquote__picture sc-ecl-picture-${this.theme}`}
          imgClass={`ecl-blockquote__image sc-ecl-blockquote-${this.theme}`}
          image={this.image}
          imageAlt={this.imageAlt}
        >
          <slot name="sources"></slot>
        </ecl-picture>
      : ''}
        <div class="ecl-blockquote__body">
          <blockquote class="ecl-blockquote__quote">
            <p class="ecl-blockquote__citation" lang={this.language}>
              <slot></slot>
            </p>
          </blockquote>
          { this.author ?
          <footer class="ecl-blockquote__attribution">
            <cite class="ecl-blockquote__author">{this.author}</cite>
          </footer> : '' }
        </div>
      </figure>
    );
  }
}
