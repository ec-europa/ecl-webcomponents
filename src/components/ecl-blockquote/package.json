{
  "name": "@ecl/web-components-blockquote",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.12.0",
  "description": "ECL Blockquote",
  "main": "ecl-blockquote.tsx",
  "private": true,
  "scripts": {
    "build:styles": "cross-env ecl-builder styles",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.11.0",
    "@ecl/theme-ec": "3.11.0",
    "@ecl/theme-eu": "3.11.0",
    "@ecl/vanilla-component-blockquote": "3.11.0",
    "cross-env": "7.0.3"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}
