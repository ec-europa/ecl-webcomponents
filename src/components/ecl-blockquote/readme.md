# ecl-blockquote



<!-- Auto Generated Below -->


## Properties

| Property     | Attribute     | Description | Type     | Default     |
| ------------ | ------------- | ----------- | -------- | ----------- |
| `author`     | `author`      |             | `string` | `undefined` |
| `colorMode`  | `color-mode`  |             | `string` | `undefined` |
| `image`      | `image`       |             | `string` | `undefined` |
| `imageAlt`   | `image-alt`   |             | `string` | `undefined` |
| `language`   | `language`    |             | `string` | `'en'`      |
| `styleClass` | `style-class` |             | `string` | `''`        |
| `theme`      | `theme`       |             | `string` | `'ec'`      |


## Dependencies

### Depends on

- [ecl-picture](../ecl-picture)

### Graph
```mermaid
graph TD;
  ecl-blockquote --> ecl-picture
  style ecl-blockquote fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
