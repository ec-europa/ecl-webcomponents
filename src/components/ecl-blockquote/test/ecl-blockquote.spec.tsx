import { newSpecPage } from '@stencil/core/testing';
import { EclBlockquote } from '../ecl-blockquote';

describe('ecl-blockquote', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclBlockquote],
      html: `<ecl-blockquote
              author="President Juncker"
              theme="ec"
              image="https://inno-ecl.s3.amazonaws.com/media/example.htmls/example.html-image-square.jpg"
            >
              An interconnected grid will help deliver the ultimate goal of the Energy Union, 
              to make sure affordable, secure and sustainable energy, and also growth across the EU.
            </ecl-blockquote>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
