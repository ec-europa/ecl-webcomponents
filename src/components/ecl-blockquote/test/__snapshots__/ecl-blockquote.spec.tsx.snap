// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-blockquote renders 1`] = `
<ecl-blockquote author="President Juncker" image="https://inno-ecl.s3.amazonaws.com/media/example.htmls/example.html-image-square.jpg" theme="ec">
  <!---->
  <figure class="ecl-blockquote">
    <div class="ecl-blockquote__body">
      <blockquote class="ecl-blockquote__quote">
        <p class="ecl-blockquote__citation">
          An interconnected grid will help deliver the ultimate goal of the Energy Union, to make sure affordable, secure and sustainable energy, and also growth across the EU.
        </p>
      </blockquote>
      <footer class="ecl-blockquote__attribution">
        <cite class="ecl-blockquote__author">
          President Juncker
        </cite>
      </footer>
    </div>
    <ecl-picture image="https://inno-ecl.s3.amazonaws.com/media/example.htmls/example.html-image-square.jpg" imgclass="ecl-blockquote__image sc-ecl-blockquote-ec" styleclass="ecl-blockquote__picture sc-ecl-picture-ec"></ecl-picture>
  </figure>
</ecl-blockquote>
`;
